nova_vm/src/ecmascript/execution/agent.rs

```diff
@@ -52,6 +52,31 @@ pub trait HostHooks: std::fmt::Debug {
     fn host_has_source_text_available(&self, func: Function) -> bool;
 }
 
+pub struct BoxedAgent {
```

Comment:
**praise:** Only allowing GC on a wrapper struct over `Agent`, which also contains the GC roots, is such a neat solution! A couple comments, though:

**question:** Is there any reason for the `agent` field to be boxed, rather than inline?

**nitpick:** The name `BoxedAgent` describes more the struct's implementation than its purpose. How about something like `GCAgent`, `GCEnabledAgent`...?